scripts/ci/setup-dev-sso.sh

```diff
@@ -67,12 +75,83 @@ run_wait_until_is_installed() {
     fi
 }
 
+setup_oauth_generic()
+{
+  set -e -o pipefail
+
+  RHSSO_URL="$1"
+
+  printf "configuring OAuth authentication for keycloak '%s'\n" "${RHSSO_URL}"
+  KEYCLOAK_SECRET="$2" envsubst < "dev-sso/openid-secret.yaml" | oc apply -f -
+
+  # Certificate used by keycloak is self-signed, we need to import and grant for it
+  printf "creating configmap with keycloak certificates"
+  oc get secrets -n openshift-ingress-operator router-ca -o jsonpath='{.data.tls\.crt}' | base64 -d > /tmp/ca.crt
+  oc create configmap ca-config-map --from-file="ca.crt=/tmp/ca.crt" -n openshift-config || true
+
+  printf "applying patch for oauths configuration\n"
+
+  oc patch oauths.config.openshift.io/cluster --type=merge --patch-file=/dev/stdin << EOF
+spec:
+  identityProviders:
+  - mappingMethod: lookup
+    name: rhd
+    openID:
+      ca:
+        name: ca-config-map
+      claims:
+        preferredUsername:
+        - preferred_username
+        email:
+        - email
+        name:
+        - username
+      clientID: kubesaw
+      clientSecret:
+        name: openid-client-secret-kubesaw
+      issuer: ${RHSSO_URL}/auth/realms/kubesaw-dev
+    type: OpenID
+EOF
+}
+
+setup_oauth_rosa()
+{
+  ISSUER_URL="$1/auth/realms/kubesaw-dev"
+  KEYCLOAK_SECRET="$2"
+  CLUSTER_NAME="$3"
+
+  printf "Setting up OAuth in ROSA cluster '%s' for issuer '%s'\n" "${CLUSTER_NAME}" "${ISSUER_URL}"
+  rosa create idp \
+    --cluster="${CLUSTER_NAME}" \
+    --type='openid' \
+    --client-id='kubesaw' \
+    --client-secret="${KEYCLOAK_SECRET}" \
+    --mapping-method='lookup' \
+    --issuer-url="${ISSUER_URL}" \
+    --email-claims='email' \
+    --name-claims='username' \
+    --username-claims='username' \
+    --name='rhd' \
+    --yes
+}
+
+setup_oauth()
+{
+  set -e
+
+  if [[ -n "$3" ]]; then
+    setup_oauth_rosa "$1" "$2" "$3"
+  else
+    setup_oauth_generic "$1" "$2"
+  fi
+}
+
 read_arguments "$@"
 
 set -e
 
 
-check_commands oc base64 openssl
+check_commands oc base64 openssl rosa
```

Comment:
We can gate this in such a way it's checked only if the target is a Rosa cluster

```suggestion
check_commands oc base64 openssl
[ -n "${CLUSTER_NAME}" ] && check_commands rosa
```

Reply:
makes sense indeed! Thanks!